lib/miq_expression/field.rb

```diff
@@ -18,6 +18,12 @@ def self.parse(field)
         parsed_params[:virtual_custom_column])
   end
 
+  def self.get_col_name(field)
+    return field.split('.').last.tr('-', '.') unless is_field?(field)
+    parsed_field = parse(field)
+    parsed_field.associations.push(parsed_field.column).join('.')
```

Comment:
Maybe we could add an instance method that encapsulates this presentation form of the field with a possibly better name than `get_col_name` (which I know is not yours :wink:). Something like `to_human` could be consistent with other parts of MiqExpression (i.e. "this is the form that we want to present in the UI"), WDYT?